docker/api/exec_api.py

```diff
@@ -65,4 +64,15 @@ def exec_start(self, exec_id, detach=False, tty=False, stream=False):
         res = self._post_json(
             self._url('/exec/{0}/start', exec_id), data=data, stream=stream
         )
+        return res
+
+    @utils.minimum_version('1.15')
+    def exec_start(self, exec_id, detach=False, tty=False, stream=False):
+        res = self._exec_start(exec_id, detach=detach, tty=tty, stream=stream)
         return self._get_result_tty(stream, res, tty)
+
+    @utils.minimum_version('1.15')
+    def exec_start_socket(self, exec_id, detach=False, tty=False,
+                          stream=False):
+        res = self._exec_start(exec_id, detach=detach, tty=tty, stream=stream)
+        return self._get_raw_response_socket(res)
```

Comment:
This does follow the pattern established by `attach()`/`attach_socket()` but I wonder if it could be just another kwarg like stream?

What do you think @shin-  ?

```python
def exec_start(self, exec_id, detach=False, tty=False, stream=False, socket=False):
    ...
    func = get_response_handler(stream, socket)
    return func(res)

def get_response_handler(stream, socket):
    if stream and socket:
        raise ValueError(...)

    if stream:
        return functools.partial(self._get_result_tty, True)
    if socket:
        return  self._get_raw_response_socket
    return  functools.partial(self._get_result_tty, False)
```